hugr-core/src/hugr/rewrite/simple_replace.rs

```diff
@@ -161,23 +195,17 @@ impl Rewrite for SimpleReplacement {
                 //
                 // Otherwise, we might disconnect other wires in `rem_inp_node`
                 // that are needed for the following iterations.
-                connect.insert((
+                h.connect(
```

Comment:
the comment above is also no longer true - but it is not clear to me how any disconnects might happen, and all tests (including those involving direct wires from input to output in the replacement) work

Reply:
This was meant to fix #1323. See #1324 / `test_half_nots`.
I'm not sure why that it's not a problem anymore, I'd have to look at the logic here for a bit.